### ISSUE TITLE
Fail to build Dockerfile: unpacking of archive failed on file /etc/hosts: cpio: rename

### ISSUE THREAD

**Issue by patrickdillon** [containers/buildah#1046]
**Description**

I am trying to build a Dockerfile. It builds ok with Docker but fails with buildah.

**Steps to reproduce the issue:**
1. Dockerfile: http://pastebin.test.redhat.com/652536
2. `podman build -t install-test .`

**Describe the results you received:**
Error logs: http://pastebin.test.redhat.com/652525

**Describe the results you expected:**
Expected it to build (works with Docker).

**Output of `rpm -q buildah` or `apt list buildah`:**

```
buildah-1.3-1.git4888163.fc28.x86_64
```

**Output of `buildah version`:**

```
Version:         1.3
Go Version:      go1.10.3
Image Spec:      1.0.0
Runtime Spec:    1.0.0
CNI Spec:        0.3.1
libcni Version:  v0.6.0
Git Commit:      4888163
Built:           Sun Aug  5 07:31:55 2018
OS/Arch:         linux/amd64
```

**Output of `cat /etc/*release`:**

```
Fedora release 28 (Twenty Eight)
NAME=Fedora
VERSION="28 (Workstation Edition)"
ID=fedora
VERSION_ID=28
PLATFORM_ID="platform:f28"
PRETTY_NAME="Fedora 28 (Workstation Edition)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:28"
HOME_URL="https://fedoraproject.org/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=28
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=28
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
Fedora release 28 (Twenty Eight)
Fedora release 28 (Twenty Eight)
```

**Output of `uname -a`:**

```
Linux localhost.localdomain 4.18.5-200.fc28.x86_64 #1 SMP Tue Sep 4 15:56:14 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Output of `cat /etc/containers/storage.conf`:**

```
# This file is is the configuration file for all tools
# that use the containers/storage library.
# See man 5 containers-storage.conf for more information
# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "overlay"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# Storage options to be passed to underlying storage drivers

# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# Path to an helper program to use for mounting the file system instead of mounting it
# directly.
#mount_program = "/usr/bin/fuse-overlayfs"

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
override_kernel_check = "true"

# mountopt specifies comma separated list of extra mount options
mountopt = "nodev"

# Remap-UIDs/GIDs is the mapping from UIDs/GIDs as they should appear inside of
# a container, to UIDs/GIDs as they should appear outside of the container, and
# the length of the range of UIDs/GIDs.  Additional mapped sets can be listed
# and will be heeded by libraries, but there are limits to the number of
# mappings which the kernel will allow when you later attempt to run a
# container.
#
# remap-uids = 0:1668442479:65536
# remap-gids = 0:1668442479:65536

# Remap-User/Group is a name which can be used to look up one or more UID/GID
# ranges in the /etc/subuid or /etc/subgid file.  Mappings are set up starting
# with an in-container ID of 0 and the a host-level ID taken from the lowest
# range that matches the specified name, and using the length of that range.
# Additional ranges are then assigned, using the ranges which specify the
# lowest host-level IDs first, to the lowest not-yet-mapped container-level ID,
# until all of the entries have been used for maps.
#
# remap-user = "storage"
# remap-group = "storage"

[storage.options.thinpool]
# Storage Options for thinpool

# autoextend_percent determines the amount by which pool needs to be
# grown. This is specified in terms of % of pool size. So a value of 20 means
# that when threshold is hit, pool will be grown by 20% of existing
# pool size.
# autoextend_percent = "20"

# autoextend_threshold determines the pool extension threshold in terms
# of percentage of pool size. For example, if threshold is 60, that means when
# pool is 60% full, threshold has been hit.
# autoextend_threshold = "80"

# basesize specifies the size to use when creating the base device, which
# limits the size of images and containers.
# basesize = "10G"

# blocksize specifies a custom blocksize to use for the thin pool.
# blocksize="64k"

# directlvm_device specifies a custom block storage device to use for the
# thin pool. Required if you setup devicemapper.
# directlvm_device = ""

# directlvm_device_force wipes device even if device already has a filesystem.
# directlvm_device_force = "True"

# fs specifies the filesystem type to use for the base device.
# fs="xfs"

# log_level sets the log level of devicemapper.
# 0: LogLevelSuppress 0 (Default)
# 2: LogLevelFatal
# 3: LogLevelErr
# 4: LogLevelWarn
# 5: LogLevelNotice
# 6: LogLevelInfo
# 7: LogLevelDebug
# log_level = "7"

# min_free_space specifies the min free space percent in a thin pool require for
# new device creation to succeed. Valid values are from 0% - 99%.
# Value 0% disables
# min_free_space = "10%"

# mkfsarg specifies extra mkfs arguments to be used when creating the base.
# device.
# mkfsarg = ""

# use_deferred_removal marks devicemapper block device for deferred removal.
# If the thinpool is in use when the driver attempts to remove it, the driver 
# tells the kernel to remove it as soon as possible. Note this does not free
# up the disk space, use deferred deletion to fully remove the thinpool.
# use_deferred_removal = "True"

# use_deferred_deletion marks thinpool device for deferred deletion.
# If the device is busy when the driver attempts to delete it, the driver
# will attempt to delete device every 30 seconds until successful.
# If the program using the driver exits, the driver will continue attempting
# to cleanup the next time the driver is used. Deferred deletion permanently
# deletes the device and all data stored in device will be lost.
# use_deferred_deletion = "True"

# xfs_nospace_max_retries specifies the maximum number of retries XFS should
# attempt to complete IO when ENOSPC (no space) error is returned by
# underlying storage device.
# xfs_nospace_max_retries = "0"

# If specified, use OSTree to deduplicate files with the overlay backend
ostree_repo = ""

# Set to skip a PRIVATE bind mount on the storage home directory.  Only supported by
# certain container storage drivers
skip_mount_home = "false"
```

**Comment by TomSweeneyRedHat** [comment]
@patrickdillon Thanks for the problem report.  This looks to be  a problem with yum not being able to find a package from the repository.  When you ran this in Docker, did you run it on the same machine that you ran the buildah bud command from?

**Comment by patrickdillon** [comment]
Yes. I ran both commands in the same shell session.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Could you see if this works for you?

**Comment by TomSweeneyRedHat** [comment]
@rhatdan ack, will give it a go Monday if not before.

**Comment by TomSweeneyRedHat** [comment]
@rhatdan FYI, I'm able to reproduce this from master, looking.

**Comment by rhatdan** [comment]
/etc/hosts is a volume mount into the container.  It is probably conflicting on that.

One big difference between the two is Docker puts /etc/hosts on a file system:
/dev/mapper/fedora-var on /etc/hosts type xfs (rw,relatime,seclabel,attr2,inode64,prjquota)

While Buildah has it on a tmpfs?
tmpfs on /etc/hosts type tmpfs (rw,nosuid,nodev,seclabel)

**Comment by TomSweeneyRedHat** [comment]
I tried running the RPM command within the container and I think you've hit the why @rhatdan I'm seeing an error on /etc/hosts there:

```
Package
       Arch   Version       Repository                                                                           Size
======================================================================================================================
Updating:
 setup noarch 2.8.71-10.el7 download-node-02.eng.bos.redhat.com_rel-eng_latest-RHEL-7_compose_Server_x86_64_os_ 166 k

Transaction Summary
======================================================================================================================
Upgrade  1 Package

Total download size: 166 k
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
setup-2.8.71-10.el7.noarch.rpm                                                                 | 166 kB  00:00:02     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : setup-2.8.71-10.el7.noarch                                                                         1/2 
Error unpacking rpm package setup-2.8.71-10.el7.noarch
error: unpacking of archive failed on file /etc/hosts: cpio: rename
  Verifying  : setup-2.8.71-10.el7.noarch                                                                         1/2 
setup-2.8.71-9.el7.noarch was supposed to be removed but is not!
  Verifying  : setup-2.8.71-9.el7.noarch                                                                          2/2 

Failed:
  setup.noarch 0:2.8.71-9.el7                               setup.noarch 0:2.8.71-10.el7                              

Complete!
```

**Comment by rhatdan** [comment]
Why this is different from Docker I don't know.

**Comment by rhatdan** [comment]
@nalind WDYT?

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Do we still have this issue?

**Comment by TomSweeneyRedHat** [comment]
IDK, I'll retest.  I know we've made changes to the way we handle /etc/host in v1.5, hopefully this got cleansed with those changed.

**Comment by TomSweeneyRedHat** [comment]
This is actually looking much better.  It got past Step 3 where it failed before and failed in Step 6 try to copy ssh config around.  @patrickdillon Could you by chance try the test again with the latest Buildah (v1.5).

```
{Removed a ton of happy output}
STEP 4: ENV ANSIBLE_STDOUT_CALLBACK yaml
STEP 5: ENV ANSIBLE_STRATEGY debug
STEP 6: ADD configs/ssh-config /root/.ssh/config
error building at step {Env:[ANSIBLE_STRATEGY=debug ANSIBLE_STDOUT_CALLBACK=yaml PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin container=oci   ANSIBLE_STDOUT_CALLBACK=yaml ANSIBLE_STRATEGY=debug] Command:add Args:[configs/ssh-config /root/.ssh/config] Flags:[] Attrs:map[] Message:ADD configs/ssh-config /root/.ssh/config Original:ADD configs/ssh-config /root/.ssh/config}: no files found matching "/tmp/configs/ssh-config": no such file or directory
# 
```

**Comment by rhatdan** [comment]
That seems unrelated to the original issue?

**Comment by TomSweeneyRedHat** [comment]
@rhatdan yep, the error that I ran into was completely unrelated to the original issue.  I'm 99.999999% sure it's due to my test system not being set up in the same way as @patrickdillon 's as I didn't have a configs/ssh-config in the environment to ADD.

**Comment by rhatdan** [comment]
Ok I will claim this is fixed in buildah 1.5/

**Comment by rhatdan** [comment]
So this is still the /etc/hosts issue?

**Comment by TomSweeneyRedHat** [comment]
@rhatdan I don't think it's /etc/hosts, but I'm not sure what's going on.  I'm not able to reproduce it on Fedora 29 with the latest Buildah from upstream.  I did note that when I ran build complained about HOSTNAME being supported only in Docker format, not sure where that picked up from.  

Could there be a difference in F28 and F29 that's causing this for @patrickdillon and not me?

Here's my tests runs both that worked although the first squawked about the HOSTNAME.

```
# cat ~/Dockerfile
FROM registry.access.redhat.com/rhel7:latest

RUN yum-config-manager \
      --add-repo http://download-node-02.eng.bos.redhat.com/rel-eng/latest-RHEL-7/compose/Server/x86_64/os/ \
      --add-repo http://pulp-read.dist.prod.ext.phx2.redhat.com/content/dist/rhel/server/7/7Server/x86_64/ansible/2.6/os/ \
      --add-repo http://pulp-read.dist.prod.ext.phx2.redhat.com/content/dist/rhel/server/7/7Server/x86_64/ose/3.10/os/ \
      &>/dev/null && \
      yum-config-manager --save \
        --setopt=\*.gpgkey="file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release" \
        --setopt=\*.gpgcheck=0 \
        --setopt=\*.skip_if_unavailable=true &>/dev/null
**OCI**
# ./buildah bud -t tom -f ~/Dockerfile .
STEP 1: FROM registry.access.redhat.com/rhel7:latest
Getting image source signatures
Copying blob sha256:23113ae36f8e9d98b1423e44673979132dec59db2805e473e931d83548b0be82
 72.21 MiB / 72.21 MiB [===================================================] 20s
Copying blob sha256:d134b18b98b0d113b7b1194a60efceaa2c06eff41386d6c14b0e44bfe557eee8
 1.19 KiB / 1.19 KiB [======================================================] 0s
Copying config sha256:5edf42cf4ed898acd4d920c02ab8c69fe4d5b744d9197aea9b4a0918d68a2a32
 6.18 KiB / 6.18 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
STEP 2: RUN yum-config-manager       --add-repo http://download-node-02.eng.bos.redhat.com/rel-eng/latest-RHEL-7/compose/Server/x86_64/os/       --add-repo http://pulp-read.dist.prod.ext.phx2.redhat.com/content/dist/rhel/server/7/7Server/x86_64/ansible/2.6/os/       --add-repo http://pulp-read.dist.prod.ext.phx2.redhat.com/content/dist/rhel/server/7/7Server/x86_64/ose/3.10/os/       &>/dev/null &&       yum-config-manager --save         --setopt=\*.gpgkey="file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release"         --setopt=\*.gpgcheck=0         --setopt=\*.skip_if_unavailable=true &>/dev/null
ERRO[0032] HOSTNAME is not supported for OCI image format, hostname 1d561c58fd2b will be ignored. Must use `docker` format 
STEP 3: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]localhost/tom:latest
Getting image source signatures
Skipping fetch of repeat blob sha256:56a763045c4544c21c458f3cd948a46384e4b12f9deacd1aede445af598f6d84
Skipping fetch of repeat blob sha256:ab9227d97750aff30bf47631468e9b69dddcdd4af6a853233d6288d05770fcf8
Copying blob sha256:3ab20ddef56902027e67a75ae5192af6781d0263525364ddd9bd120d007354fa
 473.70 KiB / 473.70 KiB [==================================================] 0s
Copying config sha256:4fe02522e3f77cde897ae31d26d1e7b234bc7fe752a000271ec9bca9cdfab22f
 3.07 KiB / 3.07 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
--> 4fe02522e3f77cde897ae31d26d1e7b234bc7fe752a000271ec9bca9cdfab22f

**DOCKER**
# ./buildah rmi -a
196d12cf6ab19273823e700516e98eb1910b03b17840f9d5509f03858484d321
5edf42cf4ed898acd4d920c02ab8c69fe4d5b744d9197aea9b4a0918d68a2a32
4fe02522e3f77cde897ae31d26d1e7b234bc7fe752a000271ec9bca9cdfab22f
[root@localhost buildah]# ./buildah rm -a
[root@localhost buildah]# ./buildah bud --format=docker -t tom -f ~/Dockerfile .
STEP 1: FROM registry.access.redhat.com/rhel7:latest
Getting image source signatures
Copying blob sha256:23113ae36f8e9d98b1423e44673979132dec59db2805e473e931d83548b0be82
 72.21 MiB / 72.21 MiB [===================================================] 21s
Copying blob sha256:d134b18b98b0d113b7b1194a60efceaa2c06eff41386d6c14b0e44bfe557eee8
 1.19 KiB / 1.19 KiB [======================================================] 0s
Copying config sha256:5edf42cf4ed898acd4d920c02ab8c69fe4d5b744d9197aea9b4a0918d68a2a32
 6.18 KiB / 6.18 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
STEP 2: RUN yum-config-manager       --add-repo http://download-node-02.eng.bos.redhat.com/rel-eng/latest-RHEL-7/compose/Server/x86_64/os/       --add-repo http://pulp-read.dist.prod.ext.phx2.redhat.com/content/dist/rhel/server/7/7Server/x86_64/ansible/2.6/os/       --add-repo http://pulp-read.dist.prod.ext.phx2.redhat.com/content/dist/rhel/server/7/7Server/x86_64/ose/3.10/os/       &>/dev/null &&       yum-config-manager --save         --setopt=\*.gpgkey="file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release"         --setopt=\*.gpgcheck=0         --setopt=\*.skip_if_unavailable=true &>/dev/null
STEP 3: COMMIT containers-storage:[overlay@/var/lib/containers/storage+/var/run/containers/storage:overlay.mountopt=nodev,overlay.override_kernel_check=true]localhost/tom:latest
Getting image source signatures
Skipping fetch of repeat blob sha256:56a763045c4544c21c458f3cd948a46384e4b12f9deacd1aede445af598f6d84
Skipping fetch of repeat blob sha256:ab9227d97750aff30bf47631468e9b69dddcdd4af6a853233d6288d05770fcf8
Copying blob sha256:08f05511c200b8a86bf04569930434f3cb1539eb343a42cff857c7b6abcd0664
 473.70 KiB / 473.70 KiB [==================================================] 0s
Copying config sha256:63d20e4cd2af78206a5ff240e1af035c37e51def749aad2cbfa0be39e48736a0
 6.29 KiB / 6.29 KiB [======================================================] 0s
Writing manifest to image destination
Storing signatures
--> 63d20e4cd2af78206a5ff240e1af035c37e51def749aad2cbfa0be39e48736a0
```

**Comment by TomSweeneyRedHat** [comment]
Oh just noted, looks like you're running as non-root, is that correct @patrickdillon ?  I ran as root and it worked fine for me.

**Comment by patrickdillon** [comment]
@TomSweeneyRedHat Apologies for the confusion: yes, sudo, progressed the build. 
My build actually still fails on step 3, but docker --no-cache fails identically, so I think there is a problem with the dockerfile or package. I'm closing this, unless you think I can provide any more useful information.

**Comment by mglantz** [comment]
I can verify this on RHEL8 HTB Snapshot 2, with podman-1.0.0-1.git82e8011.module+el8+2696+e59f0461.x86_64 and this Dockerfile:

---
FROM registry.centos.org/che-stacks/centos-stack-base

RUN sudo yum -y update setup

CMD tail -f /dev/null 
---
Which fails with:
Running transaction
  Updating   : setup-2.8.71-10.el7.noarch                                   1/2 
Error unpacking rpm package setup-2.8.71-10.el7.noarch
error: unpacking of archive failed on file /etc/hosts: cpio: rename
  Verifying  : setup-2.8.71-10.el7.noarch                                   1/2 
setup-2.8.71-9.el7.noarch was supposed to be removed but is not!
  Verifying  : setup-2.8.71-9.el7.noarch                                    2/2 

Failed:
  setup.noarch 0:2.8.71-9.el7            setup.noarch 0:2.8.71-10.el7           

Complete!
error building at step {Env:[PATH=/usr/lib/jvm/java-1.8.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin JAVA_HOME=/usr/lib/jvm/java-1.8.0 HOME=/home/user] Command:run Args:[sudo yum -y update setup] Flags:[] Attrs:map[] Message:RUN sudo yum -y update setup Original:RUN sudo yum -y update setup}: error while running runtime: exit status 1

**Comment by mglantz** [comment]
I'm replicating this both running podman as root, as a non-priviledged user and non-priviledged + sudo.

**Comment by mglantz** [comment]
Here's a sosreport from the server I tested on, if it helps.
[sosreport-rhel8htb-2019-01-23-pwjmguj.zip](https://github.com/containers/buildah/files/2789452/sosreport-rhel8htb-2019-01-23-pwjmguj.zip)

**Comment by TomSweeneyRedHat** [comment]
Thanks for the follow ups, will take a look.

**Comment by rhatdan** [comment]
@mglantz Is this repeatable?  Any chance you could try this wit the latest upstream?  Or with buildah 1.6?

**Comment by mglantz** [comment]
> @mglantz Is this repeatable? Any chance you could try this wit the latest upstream? Or with buildah 1.6?
I will try to replicate on a fresh RHEL8 snapshot2 instance today. This was with buildah-1.5-3.gite94b4f9.module+el8+2650+e6b3d617.x86_64. I'll try out 1.6, today if there is time. I've added this to RHEL8 HTB tracker BZ#1647562 so it doesn't get forgotten.

**Comment by mglantz** [comment]
> @mglantz Is this repeatable? Any chance you could try this wit the latest upstream? Or with buildah 1.6?

Yes, I can replicate this, also in latest snapshot (4) of RHEL 8 HTB. Next, to try with buildah 1.6 or upstream latest if possible.

Replication script:
yum -y install podman buildah
cat << 'EOF' >Dockerfile
FROM registry.centos.org/che-stacks/centos-stack-base

RUN sudo yum -y update setup

CMD tail -f /dev/null
EOF

podman build -t podman:test .

**Comment by mglantz** [comment]
Please note that building something without updating setup works.

[root@rhel8htb ~]# cat Dockerfile
FROM registry.centos.org/che-stacks/centos-stack-base

RUN sudo echo test

CMD tail -f /dev/null
[root@rhel8htb ~]# podman build -t podman:test .
STEP 1: FROM registry.centos.org/che-stacks/centos-stack-base
STEP 2: RUN sudo echo test
test
--> cd02e9509fef7917a209bc39c4bed9b890f5642c942579bb7b82f6b20beb6430
STEP 3: FROM cd02e9509fef7917a209bc39c4bed9b890f5642c942579bb7b82f6b20beb6430
STEP 4: CMD tail -f /dev/null
--> 4b885c81a7cdc61bb3f547c72801b77fac88b5e258ce420ae603f020541909eb
STEP 5: COMMIT podman:test

**Comment by mglantz** [comment]
@rhatdan I can replicate this with the latest build of buildah that could find in Fedora: https://src.fedoraproject.org/rpms/buildah/c/664c8a8c71697aa67bd99df5f6e2c71b8a4a8586?branch=master + RHEL8 HTB Snapshot 4 (latest).
...
Upgraded:
  buildah-1.7-5.dev.gitf1cec50.fc30.x86_64                                                                                                              

Complete!
[root@rhel8htb ~]# rpm -q buildah
buildah-1.7-5.dev.gitf1cec50.fc30.x86_64
[root@rhel8htb ~]# podman build -t podman:test .
STEP 1: FROM registry.centos.org/che-stacks/centos-stack-base
STEP 2: RUN sudo yum -y update setup
Loaded plugins: fastestmirror, ovl
Determining fastest mirrors
 * base: mirrors.glesys.net
 * extras: mirrors.glesys.net
 * updates: ftp.ember.se
base                                                     | 3.6 kB     00:00     
centos-sclo-rh                                           | 3.0 kB     00:00     
centos-sclo-sclo                                         | 2.9 kB     00:00     
extras                                                   | 3.4 kB     00:00     
updates                                                  | 3.4 kB     00:00     
(1/6): base/7/x86_64/group_gz                              | 166 kB   00:00     
(2/6): extras/7/x86_64/primary_db                          | 156 kB   00:00     
(3/6): centos-sclo-sclo/x86_64/primary_db                  | 257 kB   00:00     
(4/6): base/7/x86_64/primary_db                            | 6.0 MB   00:01     
(5/6): updates/7/x86_64/primary_db                         | 1.4 MB   00:01     
(6/6): centos-sclo-rh/x86_64/primary_db                    | 3.6 MB   00:02     
Resolving Dependencies
--> Running transaction check
---> Package setup.noarch 0:2.8.71-9.el7 will be updated
---> Package setup.noarch 0:2.8.71-10.el7 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package         Arch             Version                  Repository      Size
================================================================================
Updating:
 setup           noarch           2.8.71-10.el7            base           166 k

Transaction Summary
================================================================================
Upgrade  1 Package

Total download size: 166 k
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
setup-2.8.71-10.el7.noarch.rpm                             | 166 kB   00:00     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : setup-2.8.71-10.el7.noarch                                   1/2 
Error unpacking rpm package setup-2.8.71-10.el7.noarch
error: unpacking of archive failed on file /etc/hosts: cpio: rename
  Verifying  : setup-2.8.71-10.el7.noarch                                   1/2 
setup-2.8.71-9.el7.noarch was supposed to be removed but is not!
  Verifying  : setup-2.8.71-9.el7.noarch                                    2/2 

Failed:
  setup.noarch 0:2.8.71-9.el7            setup.noarch 0:2.8.71-10.el7           

Complete!
error building at step {Env:[PATH=/usr/lib/jvm/java-1.8.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin JAVA_HOME=/usr/lib/jvm/java-1.8.0 HOME=/home/user] Command:run Args:[sudo yum -y update setup] Flags:[] Attrs:map[] Message:RUN sudo yum -y update setup Original:RUN sudo yum -y update setup}: error while running runtime: exit status 1

**Comment by mglantz** [comment]
[podman.strace.txt.zip](https://github.com/containers/buildah/files/2792658/podman.strace.txt.zip)
If you grep for 'hosts' or 'error: unpacking of archive failed on file /etc/hosts: cpio: rename' in this strace I did on 'podman build -t podman:test .' that might give you guys a better idea of what goes on.

**Comment by mglantz** [comment]
[podman-build-debug.txt](https://github.com/containers/buildah/files/2792732/podman-build-debug.txt)
Output from 'podman --log-level debug build -t podman:test .'

**Comment by mglantz** [comment]
I forced an upgrade to podman-1.0.1-11.dev.gitc9e1f36.fc30.x86_64 (ignoring conmon dep) without any difference.

**Comment by TomSweeneyRedHat** [comment]
Thanks for the 411 and testing time @mglantz, I'll try to recreate on our end.

**Comment by mglantz** [comment]
Ehm, I found a workaround which should be able to lead you to what's wrong, I guess.
When configuring my test servers /etc/hosts with an entry for the servers hostname (in my case rhel8htb), the build passes.

```
[root@rhel8htb ~]# cat /etc/hosts
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
[root@rhel8htb ~]# vi /etc/hosts
[root@rhel8htb ~]# cat /etc/hosts
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
192.168.56.20 rhel8htb rhel8htb.localdomain
[root@rhel8htb ~]# cat Dockerfile 
FROM registry.centos.org/che-stacks/centos-stack-base

RUN sudo yum -y update setup

CMD tail -f /dev/null
[root@rhel8htb ~]# podman build -t podman:test .
STEP 1: FROM registry.centos.org/che-stacks/centos-stack-base
STEP 2: RUN sudo yum -y update setup
Loaded plugins: fastestmirror, ovl
Determining fastest mirrors
 * base: centos.mirror.far.fi
 * extras: centos.mirror.gnu.fi
 * updates: mirror.one.com
base                                                     | 3.6 kB     00:00     
centos-sclo-rh                                           | 3.0 kB     00:00     
centos-sclo-sclo                                         | 2.9 kB     00:00     
extras                                                   | 3.4 kB     00:00     
updates                                                  | 3.4 kB     00:00     
(1/6): base/7/x86_64/group_gz                              | 166 kB   00:00     
(2/6): extras/7/x86_64/primary_db                          | 156 kB   00:00     
(3/6): centos-sclo-sclo/x86_64/primary_db                  | 257 kB   00:00     
(4/6): updates/7/x86_64/primary_db                         | 1.4 MB   00:00     
(5/6): centos-sclo-rh/x86_64/primary_db                    | 3.6 MB   00:01     
(6/6): base/7/x86_64/primary_db                            | 6.0 MB   00:02     
Resolving Dependencies
--> Running transaction check
---> Package setup.noarch 0:2.8.71-9.el7 will be updated
---> Package setup.noarch 0:2.8.71-10.el7 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package         Arch             Version                  Repository      Size
================================================================================
Updating:
 setup           noarch           2.8.71-10.el7            base           166 k

Transaction Summary
================================================================================
Upgrade  1 Package

Total download size: 166 k
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
setup-2.8.71-10.el7.noarch.rpm                             | 166 kB   00:00     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : setup-2.8.71-10.el7.noarch                                   1/2 
warning: /etc/shadow created as /etc/shadow.rpmnew
  Cleanup    : setup-2.8.71-9.el7.noarch                                    2/2 
  Verifying  : setup-2.8.71-10.el7.noarch                                   1/2 
  Verifying  : setup-2.8.71-9.el7.noarch                                    2/2 

Updated:
  setup.noarch 0:2.8.71-10.el7                                                  

Complete!
--> 7c7b5f99c1b2152ffdb88b9efd55f195d35ed6f61f496a0ff1258d9df419b9c9
STEP 3: FROM 7c7b5f99c1b2152ffdb88b9efd55f195d35ed6f61f496a0ff1258d9df419b9c9
STEP 4: CMD tail -f /dev/null
--> 96968cda11b8015da0c00119a25da267ed80a86e845a8abadd8eb8133c3c8a6e
STEP 5: COMMIT podman:test
```

**Comment by mglantz** [comment]
Please note that when doing this 'work around', I was using buildah-1.5-3.gite94b4f9.module+el8+2650+e6b3d617.x86_64, podman-1.0.0-1.git82e8011.module+el8+2696+e59f0461.x86_64, runc-1.0.0-54.rc5.dev.git2abd837.module+el8+2650+e6b3d617.x86_64. The out of the box versions from RHEL 8 HTB Snapshot 4.

Reverting the /etc/hosts change on my test host, so that it does not include an entry for my hostname, does NOT seem to re-introduce the problem. I don't feel like getting smarter by the minute... but perhaps this is a clue somehow. Perhaps something was missing, which somehow got permanently created. I'll test again on a fresh install and record a list of all files before/after, to try to find out.
```
[root@rhel8htb ~]# podman image rm -f -a
7c7b5f99c1b2152ffdb88b9efd55f195d35ed6f61f496a0ff1258d9df419b9c9
96968cda11b8015da0c00119a25da267ed80a86e845a8abadd8eb8133c3c8a6e
a87f2e688c91fc6816cd51f25c3e849c361e03e442e67c428568219ab4ac75b0
[root@rhel8htb ~]# podman image list
[root@rhel8htb ~]# podman build -t podman:test .
STEP 1: FROM registry.centos.org/che-stacks/centos-stack-base
Getting image source signatures
Copying blob 205941c9c2d1: 71.25 MiB / 71.25 MiB [=========================] 27s
Copying blob 630f55a4ec45: 90.63 MiB / 90.63 MiB [=========================] 27s
Copying blob 5b9f8d29506d: 126 B / 126 B [=================================] 27s
Copying blob 032a3400ce31: 2.93 KiB / 2.93 KiB [===========================] 27s
Copying blob 223e0e3f327e: 713 B / 713 B [=================================] 27s
Copying config a87f2e688c91: 6.35 KiB / 6.35 KiB [==========================] 0s
Writing manifest to image destination
Storing signatures
STEP 2: RUN sudo yum -y update setup
Loaded plugins: fastestmirror, ovl
Determining fastest mirrors
 * base: ftp.uninett.no
 * extras: ftp.uninett.no
 * updates: ftp.ember.se
base                                                     | 3.6 kB     00:00     
centos-sclo-rh                                           | 3.0 kB     00:00     
centos-sclo-sclo                                         | 2.9 kB     00:00     
extras                                                   | 3.4 kB     00:00     
updates                                                  | 3.4 kB     00:00     
(1/6): base/7/x86_64/group_gz                              | 166 kB   00:00     
(2/6): extras/7/x86_64/primary_db                          | 156 kB   00:00     
(3/6): centos-sclo-sclo/x86_64/primary_db                  | 257 kB   00:00     
(4/6): base/7/x86_64/primary_db                            | 6.0 MB   00:01     
(5/6): updates/7/x86_64/primary_db                         | 1.4 MB   00:00     
(6/6): centos-sclo-rh/x86_64/primary_db                    | 3.6 MB   00:02     
Resolving Dependencies
--> Running transaction check
---> Package setup.noarch 0:2.8.71-9.el7 will be updated
---> Package setup.noarch 0:2.8.71-10.el7 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package         Arch             Version                  Repository      Size
================================================================================
Updating:
 setup           noarch           2.8.71-10.el7            base           166 k

Transaction Summary
================================================================================
Upgrade  1 Package

Total download size: 166 k
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
setup-2.8.71-10.el7.noarch.rpm                             | 166 kB   00:00     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : setup-2.8.71-10.el7.noarch                                   1/2 
warning: /etc/shadow created as /etc/shadow.rpmnew
  Cleanup    : setup-2.8.71-9.el7.noarch                                    2/2 
  Verifying  : setup-2.8.71-10.el7.noarch                                   1/2 
  Verifying  : setup-2.8.71-9.el7.noarch                                    2/2 

Updated:
  setup.noarch 0:2.8.71-10.el7                                                  

Complete!
--> b0a3d4432aab07a1e958aba4ec7dc6ce97da393301d2be5aca9589f8c2487195
STEP 3: FROM b0a3d4432aab07a1e958aba4ec7dc6ce97da393301d2be5aca9589f8c2487195
STEP 4: CMD tail -f /dev/null
--> 15ab7c85cdaca179ca789d5bc8d9038a7a32eadd2d981d3ac2c32c914fde67bb
STEP 5: COMMIT podman:test
[root@rhel8htb ~]# 
```

**Comment by mglantz** [comment]
Perhaps a wild goose chase, but difference from an initial failed run on a newly installed server, to the first successful run, is:
```
/var/lib/containers/storage/overlay/backingFsBlockDev
```
And
```
/var/lib/containers/storage/overlay/l/FJFIQ2BZGRFR7GJAMCF3763
/var/lib/containers/storage/overlay/720ac4c54619badde0815b53b
...
/var/lib/containers/storage/overlay/c5219d6d53f9b7f0ad9a79296
...
/var/lib/containers/storage/overlay-layers/720ac4c54619badde0
...
/var/lib/containers/storage/overlay-images/aaba8f7a6ec631f281
```

Creation of backingFsBlockDev seems to live in: vendor/github.com/containers/storage/drivers/quota/projectquota.go
```
// projectquota.go - implements XFS project quota controls
// for setting quota limits on a newly created directory.
// It currently supports the legacy XFS specific ioctls.
```

**Comment by mglantz** [comment]
I did a replicator where I install two new RHEL 8 HTB snap4 instances, and only once run:
podman build --log-level debug -t podman:test .
On one machine, I implemented the work around, on one not, before running the commands, so the output should be very similar and point at what is different. 

Here's the debug output, attached. Perhaps it contains something to move this forward.
podman-build-ok.txt and podman-build-fail.txt containers the result of the command, in full. In the *-debug-msg.txt files I filtered out only debug messages.

[podman-build-fail-debug-msg.txt](https://github.com/containers/buildah/files/2793711/podman-build-fail-debug-msg.txt)
[podman-build-ok-debug-msg.txt](https://github.com/containers/buildah/files/2793712/podman-build-ok-debug-msg.txt)
[podman-build-ok.txt](https://github.com/containers/buildah/files/2793713/podman-build-ok.txt)
[podman-build-fail.txt](https://github.com/containers/buildah/files/2793714/podman-build-fail.txt)

**Comment by mglantz** [comment]
Looking at the differences between ok build and failed build, and specifically on DEBU[n] messages, the failed build displays messages about "DEBU[0002] Detected compression format gzip" which the ok build does not. Perhaps that is clue that the problem is triggered by a race condition. I'll be online tomorrow if you need more info from my test beds.

**Comment by mglantz** [comment]
If it works on docker, but not on podman, and as I could test - the problem only appears if hostname does not have an entry in /etc/hosts on the machine you build on - then perhaps the issue is related to https://github.com/containers/image/issues/72. Just a thought.

**Comment by mglantz** [comment]
Worth to note is that there are a lot of systems without a hostname entry in /etc/hosts, so this seems like something a lot of people will hit. At least on RHEL 8, where I can replicate this at will on a default install (RHEL does not by default create a hostname entry in /etc/hosts, probably also the case for most Linux OSs).

**Comment by TomSweeneyRedHat** [comment]
@mglantz thanks for all the follow ups.  I'm able to replicate this on F29 with Buildah 1.6 using the Dockerfile that you supplied 

```
FROM registry.centos.org/che-stacks/centos-stack-base
RUN sudo yum -y update setup
CMD tail -f /dev/null
```

The tail end of the error is from dnf not being happy, likely as you pointed out due to /etc/hosts being out of whack.

```
{snipped happy dnf output}
Running transaction
  Updating   : setup-2.8.71-10.el7.noarch                                                                                                                     1/2 
Error unpacking rpm package setup-2.8.71-10.el7.noarch
error: unpacking of archive failed on file /etc/hosts: cpio: rename
  Verifying  : setup-2.8.71-10.el7.noarch                                                                                                                     1/2 
setup-2.8.71-9.el7.noarch was supposed to be removed but is not!
  Verifying  : setup-2.8.71-9.el7.noarch                                                                                                                      2/2 

Failed:
  setup.noarch 0:2.8.71-9.el7                                                     setup.noarch 0:2.8.71-10.el7                                                    

Complete!
error building at step {Env:[PATH=/usr/lib/jvm/java-1.8.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin JAVA_HOME=/usr/lib/jvm/java-1.8.0 HOME=/home/user] Command:run Args:[sudo yum -y update setup] Flags:[] Attrs:map[] Message:RUN sudo yum -y update setup Original:RUN sudo yum -y update setup}: error while running runtime: exit status 1
```

Now to try to figure out the why, although this seems to be vaguely familar, I thought we'd an issue like this back around the 1.3 time frame....

**Comment by mglantz** [comment]
> Now to try to figure out the why, although this seems to be vaguely familar, I thought we'd an issue like this back around the 1.3 time frame....
Yeah @TomSweeneyRedHat, in 1.3, there was a fix for 'Force ownership of /etc/hosts and /etc/resolv.conf to 0:0'.

**Comment by TomSweeneyRedHat** [comment]
@rhatdan @nalind, I'm suspecting we might have a systemd error.  I found the same issue running on Buildah 1.5, so this is not a new to that issue.  

I've pulled the Dockerfile out of the equation and am simply doing:
```
# ctr=$(buildah from registry.centos.org/che-stacks/centos-stack-base)
# buildah run $ctr /bin/bash
$ sudo -y install setup
{removed happy dnf output}
Running transaction
  Updating   : setup-2.8.71-10.el7.noarch                                                                                                                     1/2 
Error unpacking rpm package setup-2.8.71-10.el7.noarch
error: unpacking of archive failed on file /etc/hosts: cpio: rename
  Verifying  : setup-2.8.71-10.el7.noarch                                                                                                                     1/2 
setup-2.8.71-9.el7.noarch was supposed to be removed but is not!
  Verifying  : setup-2.8.71-9.el7.noarch                                                                                                                      2/2 

Failed:
  setup.noarch 0:2.8.71-9.el7                                                     setup.noarch 0:2.8.71-10.el7
```
If I try to remove setup:

$ sudo yum remove -y setup
--> Running transaction check
---> Package openssh-clients.x86_64 0:7.4p1-16.el7 will be erased
--> Finished Dependency Resolution
Error: Trying to remove "systemd", which is protected
```

The /etc/hosts file in the container looks OK to me.

```
$ ls -alF /etc/hosts
-rw-r--r--. 1 root root 158 Jul 13  2018 /etc/hosts
```

So I'm thinking this might be a systemd issue.  I don't see any evidence of it being a selinux issue and running with that turned off has no effect.

Thoughts?

**Comment by rhatdan** [comment]
After reading all of these remarks, is the solution to the problem populating the hostname within the /etc/hosts that gets mounted into the container?

**Comment by mglantz** [comment]
> After reading all of these remarks, is the solution to the problem populating the hostname within the /etc/hosts that gets mounted into the container?

Not sure how it works, but, putting the hostname into /etc/hosts of the OS instance running the container solves the problem.

**Comment by rhatdan** [comment]
The problem, I think is that the /etc/hosts file is a mount point, and the setup install is trying to rename it.
```
# buildah run alpine-working-container sh
/ # mv /etc/hosts /etc/hosts1
mv: can't rename '/etc/hosts': Resource busy
```

Same thing happens with Docker and Podman.

**Comment by rhatdan** [comment]
error: unpacking of archive failed on file /etc/hosts: cpio: rename

**Comment by TomSweeneyRedHat** [comment]
@rhatdan I agree with the /etc/host being a mount point and setup trying to rename it as the probable issue.  Not sure how to get around that though.

**Comment by mglantz** [comment]
Can it perhaps be like this?

The setup RPM contains /etc/hosts. AFAIKS, the RPM includes no scripts which does things with /etc/hosts and RPM should not replace any existing files, by default at least. So, why does setup then touch /etc/hosts? I suppose that is because it does not see /etc/hosts, and tries to put the file in place - which then does not work.

But, when I add the FQDN to /etc/hosts, on the server which builds the container, setup inside of the container sees /etc/hosts, and then it does not try to replace or add the file - therefor there are no problems.

**Comment by mglantz** [comment]
@rhatdan strangely enough, I now found this:

If I build the container with below:
```
FROM registry.centos.org/che-stacks/centos-stack-base
RUN if [ -f /etc/hosts]; then echo no; else yum -y update setup; fi
CMD tail -f /dev/null
```
Then the RPM does not get installed and a container get's built without issues. So, if the problem is that setup cannot see /etc/hosts, it has some other view than when that evaluation is run. If I then re-build the container with:
```
FROM registry.centos.org/che-stacks/centos-stack-base
RUN yum -y update setup
CMD tail -f /dev/null
```
There are no problems and setup get's updated, even without doing anything to /etc/hosts on my build host.

Please note that once I've gotten past this issue, I cannot replicate the problem anymore.

**Comment by rhatdan** [comment]
Looking at setup.conf, it lists /etc/hosts as
%config(noreplace) /etc/host.conf

Which means it checks to see if the config has been modified from the default, and if so it does not replace it.  If the /etc/host.conf has not been changed then it will replace it.  

This is why when you modify the /etc/hosts from the system setup.conf does not change it.

What we could do is modify the hosts file that we are mounting into the container that the hosts came from inside of a container, and then setup.conf will no longer attempt to replace it.

**Comment by rhatdan** [comment]
This should be fixed now.
